sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestUtils.java

```diff
@@ -144,22 +150,25 @@ private TestUtils() {
     }
 
     /**
-     * Locates the root of the current repo by finding the eng folder's parent.
+     * Locates the root of the current repo until the provided folder's parent.
+     *
+     * @param testClassPath the test class path
+     * @param resolveFolder the folder parent to resolve the path until
      * @return The {@link Path} to the root of the repo.
-     * @throws RuntimeException The eng folder could not be located.
+     * @throws RuntimeException The specified folder could not be located.
      */
-    public static Path getRepoRoot() {
-        Path path = getRecordFolder().toPath();
+    public static Path getRepoRootResolveUntil(Path testClassPath, String resolveFolder) {
```

Comment:
We should add some logic to prevent file walking from escaping outside of the azure-sdk-for-java root.